octobot_trading/traders/trader.py

```diff
@@ -145,48 +154,122 @@ async def _create_not_loaded_order(self, new_order: Order, portfolio) -> Order:
 
         return new_order
 
-    async def cancel_order(self, order: Order):
+    async def close_filled_order(self, order):
+        """
+        Closes the given filled order starting buy cancelling its linked orders, updating the portfolio, creating
+        the new trade and finally removing the order from order manager. Does not update the order attributes.
+        :param order: Already filled order
+        :return: None
+        """
+        # Cancel linked orders
+        for linked_order in order.linked_orders:
+            await self.cancel_order(linked_order, ignored_order=order)
+
+        # update portfolio with filled order
+        async with self.exchange_manager.exchange_personal_data.get_order_portfolio(order).lock:
+            await self.exchange_manager.exchange_personal_data.handle_portfolio_update_from_order(order)
+
+        # add to trade history and notify
+        await self.exchange_manager.exchange_personal_data.handle_trade_instance_update(
+            create_trade_from_order(order))
+
+        # remove order from open_orders
+        self.exchange_manager.exchange_personal_data.orders_manager.remove_order_instance(order)
```

Comment:
What do you think of an `remove_order_instance` or `remove_order` in  ExchangePersonnalData ?

Reply:
Not sure we should add it: ExchangePersonnalData only does data update with additional processing (channel publish etc), adding a proxy just for this method would make it confusing regarding what is getting updated I'm affraid

Reply:
Right :+1: